From 8be8dc74799fe7c0e09dfa53aa41e954ffba291c Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Fri, 12 Jul 2019 11:43:03 +0100
Subject: [PATCH] bcm2838.dtsi: Use BCM2711 PCIe compatible string

The BCM2711 PCIe controller has a limited address range in the B0
silicon, and the driver uses a compatible string to identify the
limitation. The current Pi 4 firmware will override the compatible
string if it detects a downstream DTB and it is running on a newer
revision but set the default value to enable the workaround for
backwards-compatibility with old firmware.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm2838.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/bcm2838.dtsi
+++ b/arch/arm/boot/dts/bcm2838.dtsi
@@ -314,7 +314,8 @@
 			#interrupt-cells = <1>;
 			#size-cells = <2>;
 			bus-range = <0x0 0x01>;
-			compatible = "brcm,bcm7211-pcie", "brcm,bcm7445-pcie",
+			compatible = "brcm,bcm2711b0-pcie", // Safe value
+				     "brcm,bcm2711-pcie",
 				     "brcm,pci-plat-dev";
 			max-link-speed = <2>;
 			tot-num-pcie = <1>;
